emt4j-common/src/main/java/org/eclipse/emt4j/common/staticanalysis/InvokerImpl.java

```diff
@@ -0,0 +1,122 @@
+/********************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+package org.eclipse.emt4j.common.staticanalysis;
+
+import org.eclipse.emt4j.common.staticanalysis.impl.CastArraysAsListToArrayAnalyzer;
+import org.eclipse.emt4j.common.staticanalysis.impl.CastSystemClassLoaderToURLClassLoaderAnalyzer;
+import org.eclipse.emt4j.common.staticanalysis.impl.PatternCompileAnalyzer;
+import soot.G;
+import soot.Scene;
+import soot.SootClass;
+import soot.options.Options;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+
+class InvokerImpl implements Invoker {
+
+    private final File directory;
+
+    private final Map<String, Analyzer> analyzerMap = new HashMap<>();
+
+    InvokerImpl() throws IOException {
+
+        directory = Files.createTempDirectory("files-for-static-analysis-" + Thread.currentThread().getId() + "-").toFile();
+
+        System.out.println(directory.getAbsolutePath());
+
+        directory.deleteOnExit();
+
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                //noinspection resource
+                Files.walk(directory.toPath())
+                     .sorted(Comparator.reverseOrder())
+                     .map(Path::toFile)
+                     .forEach(File::delete);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }));
```

Comment:
The temporary directory created by `createTempDirectory` will be deleted automatically. Don't need to manually delete them.

Reply:
According to the document of this API, it is not said that the directory will be automatically deleted after the process exits, and we need to do this manually.

![image](https://user-images.githubusercontent.com/33491035/220060267-af3e0de1-319e-4447-bb8e-161178a5db58.png)